model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -275,12 +277,48 @@ class ModelReplicationServer(val repositoriesManager: RepositoriesManager) {
     }
 
     private suspend fun ApplicationCall.respondDelta(versionHash: String, baseVersionHash: String?) {
-        val delta = VersionDelta(
-            versionHash,
-            baseVersionHash,
-            objectsMap = repositoriesManager.computeDelta(versionHash, baseVersionHash),
-        )
-        delta.checkObjectHashes()
-        respond(delta)
+        when (val format = request.queryParameters["format"]) {
```

Comment:
What about using content negotiation instead to use more standard features?

Reply:
Not sure how that would look like.

Reply:
Basically, we could defined different MIME types and request them using the "Accept" HTTP header instead of the query parameter.

Reply:
There is now a new MIME type `application/x-modelix-objects`. The client sends an `Accept: application/x-modelix-objects` header to the server and the server responds with a `Content-Type: application/x-modelix-objects` header.

If the client accepts `application/json`, but not `application/x-modelix-objects` (which old client versions do), the server responds with as it did before.
If the client doesn't accept `application/json` (the user probably opened the URL in the browser) the server responds with the new format.